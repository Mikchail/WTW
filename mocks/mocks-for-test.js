

const movies = [{
  id: 2,
  title: `Bohemian Rhapsody`,
  cardImg: `../../img/bohemian-rhapsody.jpg`,
  genre: `Drama`,
  year: 2000,
  backgroundImg: `img/bg-the-grand-budapest-hotel.jpg`,
  posterImg: `img/the-grand-budapest-hotel-poster.jpg`,
  rating: 5,
  ratingCount: 240,
  description: `In the 1930s, theher murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`],
  preview: `https://upload.wikimedia.org`,
  videoLink: `https://download.blender4`,
  runTime: 99,
  isFavorite: true,
}, {
  id: 3,
  title: `Macbeth`,
  cardImg: `../../img/macbeth.jpg`,
  genre: `Drama`,
  year: 2002,
  backgroundImg: `img/bg-the-grand-budapest-hotel.jpg`,
  posterImg: `img/the-grand-budapest-hotel-poster.jpg`,
  rating: 2,
  ratingCount: 240,
  description: `In the 1930s`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`],
  preview: `https://480p.mp4`,
  videoLink: `https://480p.mp4`,
  runTime: 200,
  isFavorite: true,
}, {
  id: 4,
  title: `Aviator`,
  cardImg: `../../img/aviator.jpg`,
  genre: `Comedy`,
  year: 1988,
  backgroundImg: `img/bg-the-grand-budapest-hotel.jpg`,
  posterImg: `img/the-grand-budapest-hotel-poster.jpg`,
  rating: 7,
  ratingCount: 188,
  description: `in her murder.`,
  director: `Leo Dicaprio`,
  starring: [`Bill Murray`],
  preview: `400p.ogv.360p.webm`,
  videoLink: `mp4`,
  runTime: 188,
  isFavorite: true,
}];
const comments = [{
  "id": 1,
  "user": {
    "id": 4,
    "name": `Kate Muir`
  },
  "rating": 8.9,
  "comment": `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
  "date": `2019-05-08T14:13:56.569Z`
}, {
  "id": 2,
  "user": {
    "id": 4,
    "name": `Kate Muir`
  },
  "rating": 8.9,
  "comment": `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
  "date": `2019-05-08T14:13:56.569Z`
}];
const newMovie = {
  id: 2,
  title: `Bohemian Rhapsody`,
  cardImg: `../../img/bohemian-rhapsody.jpg`,
  genre: `Drama`,
  year: 2000,
  backgroundImg: `img/bg-the-grand-budapest-hotel.jpg`,
  posterImg: `img/the-grand-budapest-hotel-poster.jpg`,
  rating: 5,
  ratingCount: 240,
  description: `In the 1930s, theher murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`],
  preview: `https://upload.wikimedia.org`,
  videoLink: `https://download.blender4`,
  runTime: 99,
  isFavorite: false,
};


const newMovies = [{
  id: 2,
  title: `Bohemian Rhapsody`,
  cardImg: `../../img/bohemian-rhapsody.jpg`,
  genre: `Drama`,
  year: 2000,
  backgroundImg: `img/bg-the-grand-budapest-hotel.jpg`,
  posterImg: `img/the-grand-budapest-hotel-poster.jpg`,
  rating: 5,
  ratingCount: 240,
  description: `In the 1930s, theher murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`],
  preview: `https://upload.wikimedia.org`,
  videoLink: `https://download.blender4`,
  runTime: 99,
  isFavorite: false,
}, {
  id: 3,
  title: `Macbeth`,
  cardImg: `../../img/macbeth.jpg`,
  genre: `Drama`,
  year: 2002,
  backgroundImg: `img/bg-the-grand-budapest-hotel.jpg`,
  posterImg: `img/the-grand-budapest-hotel-poster.jpg`,
  rating: 2,
  ratingCount: 240,
  description: `In the 1930s`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`],
  preview: `https://480p.mp4`,
  videoLink: `https://480p.mp4`,
  runTime: 200,
  isFavorite: true,
}, {
  id: 4,
  title: `Aviator`,
  cardImg: `../../img/aviator.jpg`,
  genre: `Comedy`,
  year: 1988,
  backgroundImg: `img/bg-the-grand-budapest-hotel.jpg`,
  posterImg: `img/the-grand-budapest-hotel-poster.jpg`,
  rating: 7,
  ratingCount: 188,
  description: `in her murder.`,
  director: `Leo Dicaprio`,
  starring: [`Bill Murray`],
  preview: `400p.ogv.360p.webm`,
  videoLink: `mp4`,
  runTime: 188,
  isFavorite: true,
}];
const recivedMovies = [
  {
    "actors": undefined,
    "backgroundColor": undefined,
    "description": undefined,
    "genre": undefined,
    "id": undefined,
    "isFavorite": undefined,
    "movieCoverSrc": undefined,
    "numberVotes": undefined,
    "posterSrc": undefined,
    "previewVideoLink": undefined,
    "producer": undefined,
    "rating": undefined,
    "runTime": undefined,
    "screenshotSrc": undefined,
    "title": undefined,
    "videoSrc": undefined,
    "yearRelease": undefined,
  },
];

export {movies , newMovie,newMovies,recivedMovies,comments};
