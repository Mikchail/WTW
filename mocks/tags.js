const tags = [
  `All genres`,
  `Comedies`,
  `Crime`,
  `Documentary`,
  `Dramas`,
  `Horror`,
  `Kids & Family`,
  `Romance`,
  `Sci-Fi`,
  `Thrillers`,
];

export {tags};
