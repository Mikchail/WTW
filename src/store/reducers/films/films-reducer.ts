import { extend } from "../../../utils";
import { GET_FILMS, LOADED_FILMS } from "../../actions/films-actions";

const initialState = {
  films: [],
  isLoading: false,
  isError: false,
}

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case GET_FILMS:
      return state;
    case LOADED_FILMS:
      return extend(state, {films: action.payload, isLoading: true});
    default:
      return state;
  }
};
