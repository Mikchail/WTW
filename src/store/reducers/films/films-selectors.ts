import NameSpace from '../../name-space';
import {createSelector} from 'reselect';
import { RootState } from '../root-reducer';

export const getIsLoading = (state: RootState) => {
  return state[NameSpace.FILMS].isLoading;
};

export const getFilmById = (state: RootState, props: {selectedID: number}) => {
  const film = getFilms(state).find((it) => it.id === props.selectedID);
  return film;
};

export const getSimilarFilms = (state: RootState, id: number) => {
  const filmById = getFilmById(state, {selectedID: id});
  const films = getFilms(state).filter((film) => {
    return Boolean(film.genre === filmById.genre && film.id !== id);
  });
  return films;
};

export const getFilms = (state: RootState) => {
  return state[NameSpace.FILMS].films;
};

export const getTags = createSelector(getFilms, (films) => {
  return [
    ...new Set(
      films.map((f) => {
        return f.genre;
      })
    ),
  ];
});
export const getSelectFilm = (state: RootState, id: number) => {
  return state[NameSpace.FILMS].films.find((it) => it.id === id);
};
