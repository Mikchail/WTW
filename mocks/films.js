import { getRandomNumber, getRandomElement, encodeText } from '../src/utils';
import { tags as genres } from './tags';
import { articles } from './articles';

const titles = articles.map((item) => {
    return item.title;
});
const stringForDescription = [
    `Beautiful & luxurious apartment at great location`,
    `Nice, cozy, wood and stone place`,
    `Wood and stone place`,
    `Nice, cozy, warm big bed apartment`,
    `Beautiful & luxurious apartment at great location`,
    `Nice, cozy, wood and stone place`,
    `Wood and stone place`,
    `Nice, cozy, warm big bed apartment`,
];

const randomDate = () => {
    const min = new Date(1997, 2, 2).getTime();
    const max = new Date(2018, 2, 2).getTime();
    const r = getRandomNumber(min, max);
    return new Date(r);
};

const getMovies = (titles) => {
    return titles.reduce((acc, movie) => {
        console.log(acc);
        acc.push({
            title: movie,
            genre: getRandomElement(genres),
            rating: Math.floor(Math.random() * 5),
            descriptin: `${getRandomElement(
        stringForDescription
      )} ${getRandomElement(stringForDescription)} ${getRandomElement(stringForDescription)}`,
            year: randomDate(),
            src: encodeText(movie),
        });
        return acc;
    }, []);
};
const films = getMovies(titles);
export { films };