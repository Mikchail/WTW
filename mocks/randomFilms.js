import {getRandomNumber, getRandomElement, encodeText} from '../src/utils';
import {tags as genres} from './tags';
import {articles} from './articles';

const titles = articles.map((item) => {
  return item.title;
});

const stringForDescription = [
  `Beautiful & luxurious apartment at great location`,
  `Nice, cozy, wood and stone place`,
  `Wood and stone place`,
  `Nice, cozy, warm big bed apartment`,
  `Beautiful & luxurious apartment at great location`,
  `Nice, cozy, wood and stone place`,
  `Wood and stone place`,
  `Nice, cozy, warm big bed apartment`,
];

const randomDate = () => {
  const min = new Date(1997, 2, 2).getTime();
  const max = new Date(2018, 2, 2).getTime();
  const r = getRandomNumber(min, max);
  return new Date(r);
};

const getMovies = (array) => {
  return array.reduce((acc, movie, index) => {
    acc.push({
      id: index,
      title: movie,
      srcMovie: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
      genre: getRandomElement(genres),
      rating: Math.floor(Math.random() * 5),
      countRating: 240,
      descriptin: `${getRandomElement(stringForDescription)} ${getRandomElement(
        stringForDescription
      )} ${getRandomElement(stringForDescription)}`,
      year: randomDate().getFullYear(),
      src: `/img/${encodeText(movie)}.jpg`,
      director: `Wes Andreson`,
      starring: [
        `Bill Murray`,
        `Edward Norton`,
        `Jude Law`,
        `Willem Dafoe`,
        `and other`,
      ],
      reviews: [
        {
          description: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of
                        one of the director's funniest and most exquisitely designed movies in years.`,
          author: `Kate Muir one`,
          date: `December 24, 2016`,
          rating: `8.9`,
        },
        {
          description: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of
                        one of the director's funniest and most exquisitely designed movies in years.`,
          author: `Kate Muir one one`,
          date: `December 24, 2016`,
        },
        {
          description: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of
                        one of the director's funniest and most exquisitely designed movies in years.`,
          author: `Kate Muir two`,
          date: `December 24, 2016`,
        },
        {
          description: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of
                        one of the director's funniest and most exquisitely designed movies in years.`,
          author: `Kate Muir two two`,
          date: `December 24, 2016`,
        },
        {
          description: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of
                        one of the director's funniest and most exquisitely designed movies in years.`,
          author: `Kate Muir one one`,
          date: `December 24, 2016`,
        },
        {
          description: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of
                        one of the director's funniest and most exquisitely designed movies in years.`,
          author: `Kate Muir two`,
          date: `December 24, 2016`,
        },
        {
          description: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of
                        one of the director's funniest and most exquisitely designed movies in years.`,
          author: `Kate Muir two two`,
          date: `December 24, 2016`,
        },
        {
          description: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of
                        one of the director's funniest and most exquisitely designed movies in years.`,
          author: `Kate Muir one one`,
          date: `December 24, 2016`,
        },
        {
          description: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of
                        one of the director's funniest and most exquisitely designed movies in years.`,
          author: `Kate Muir two`,
          date: `December 24, 2016`,
        },
        {
          description: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of
                        one of the director's funniest and most exquisitely designed movies in years.`,
          author: `Kate Muir two two`,
          date: `December 24, 2016`,
        },
      ],
    });
    return acc;
  }, []);
};
const films = getMovies(titles);
export {films};
