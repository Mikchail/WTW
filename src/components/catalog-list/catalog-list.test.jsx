import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import {films} from '../../../mocks/films';
import CatalogList from './catalog-list';
import {Provider} from 'react-redux';
const mockStore = configureStore([]);

describe(`CatalogList is testing`, () => {
  it(`Renderer Catalog with props`, () => {
    const store = mockStore({});
    const history = {};
    const onSelectedFilm = () => {};

    const tree = renderer
      .create(
        <Provider store={store}>
          <CatalogList history={history} films={films} onSelectedFilm={onSelectedFilm} />
        </Provider>,
        {
          createNodeMock: () => {
            return {};
          },
        }
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
