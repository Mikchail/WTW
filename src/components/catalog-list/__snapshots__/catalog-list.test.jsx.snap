// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatalogList is testing Renderer Catalog with props 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        poster="/img/fantastic-beasts--the-crimes-of-grindelwald.jpg"
        width="280"
      >
        your browser doesn\`t support embedded videos
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#"
        onClick={[Function]}
        to="/player/0"
      >
        Fantastic Beasts: The Crimes of Grindelwald
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        poster="/img/bohemian-rhapsody.jpg"
        width="280"
      >
        your browser doesn\`t support embedded videos
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#"
        onClick={[Function]}
        to="/player/1"
      >
        Bohemian Rhapsody
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        poster="/img/what-we-do-in-the-shadows.jpg"
        width="280"
      >
        your browser doesn\`t support embedded videos
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#"
        onClick={[Function]}
        to="/player/2"
      >
        What We Do in the Shadows
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        poster="/img/revenant.jpg"
        width="280"
      >
        your browser doesn\`t support embedded videos
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#"
        onClick={[Function]}
        to="/player/3"
      >
        Revenant
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        poster="/img/johnny-english.jpg"
        width="280"
      >
        your browser doesn\`t support embedded videos
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#"
        onClick={[Function]}
        to="/player/4"
      >
        Johnny English
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        poster="/img/shutter-island.jpg"
        width="280"
      >
        your browser doesn\`t support embedded videos
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#"
        onClick={[Function]}
        to="/player/5"
      >
        Shutter Island
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        poster="/img/pulp-fiction.jpg"
        width="280"
      >
        your browser doesn\`t support embedded videos
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#"
        onClick={[Function]}
        to="/player/6"
      >
        Pulp Fiction
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        poster="/img/no-country-for-old-men.jpg"
        width="280"
      >
        your browser doesn\`t support embedded videos
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#"
        onClick={[Function]}
        to="/player/7"
      >
        No Country for Old Men
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        poster="/img/snatch.jpg"
        width="280"
      >
        your browser doesn\`t support embedded videos
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#"
        onClick={[Function]}
        to="/player/8"
      >
        Snatch
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        poster="/img/moonrise-kingdom.jpg"
        width="280"
      >
        your browser doesn\`t support embedded videos
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#"
        onClick={[Function]}
        to="/player/9"
      >
        Moonrise Kingdom
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        poster="/img/seven-years-in-tibet.jpg"
        width="280"
      >
        your browser doesn\`t support embedded videos
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#"
        onClick={[Function]}
        to="/player/10"
      >
        Seven Years in Tibet
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        poster="/img/midnight-special.jpg"
        width="280"
      >
        your browser doesn\`t support embedded videos
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#"
        onClick={[Function]}
        to="/player/11"
      >
        Midnight Special
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        poster="/img/war-of-the-worlds.jpg"
        width="280"
      >
        your browser doesn\`t support embedded videos
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#"
        onClick={[Function]}
        to="/player/12"
      >
        War of the Worlds
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        poster="/img/dardjeeling-limited.jpg"
        width="280"
      >
        your browser doesn\`t support embedded videos
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#"
        onClick={[Function]}
        to="/player/13"
      >
        Dardjeeling Limited
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        poster="/img/orlando.jpg"
        width="280"
      >
        your browser doesn\`t support embedded videos
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#"
        onClick={[Function]}
        to="/player/14"
      >
        Orlando
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        poster="/img/mindhunter.jpg"
        width="280"
      >
        your browser doesn\`t support embedded videos
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#"
        onClick={[Function]}
        to="/player/15"
      >
        Mindhunter
      </a>
    </h3>
  </article>
</div>
`;
